parlai/core/agents.py

```diff
@@ -495,6 +495,12 @@ def load_agent_module(opt):
                         "previously: {} )]".format(k, v, new_opt.get(k, None))
                     )
                 new_opt[k] = v
+
+        model_class = get_agent_module(new_opt['model'])
+
+        if hasattr(model_class, 'upgrade_opt'):
```

Comment:
I think we need to separate this into another PR.

Reply:
ok